tests/unit/repo/test_ua_manager.py

```diff
@@ -0,0 +1,70 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2021 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import pytest
+
+from snapcraft.internal.repo import ua_manager
+
+
+def test_ua_manager(fake_process):
+    ua_token = "test-ua-token"
+
+    fake_process.register_subprocess(
+        ["sudo", "ua", "status", "--format", "json"],
+        stdout='{"attached": false, "_doc": "Content provided in json response is currently considered Experimental and may change", "techSupportLevel": "n/a", "origin": null, "expires": "n/a", "services": [{"name": "cc-eal", "description": "Common Criteria EAL2 Provisioning Packages", "available": "no"}, {"name": "esm-infra", "description": "UA Infra: Extended Security Maintenance", "available": "yes"}, {"name": "fips", "description": "NIST-certified FIPS modules", "available": "no"}, {"name": "fips-updates", "description": "Uncertified security updates to FIPS modules", "available": "no"}, {"name": "livepatch", "description": "Canonical Livepatch service", "available": "yes"}]}',
```

Comment:
Please move out this BIG line, so each test is not that ugly.

Define only one line outside, with a small `{whatever}` template so then in each test you can actually build the needed one with the bits that change from test to test.

This way also it's more clear to the reader what is changing from test to test.

Reply:
this could just be a local pytest.fixture that returns a dict that you end up modifying from test to test

Reply:
all i really need is {"attached": false}, the rest of there is just for real data. will break it up or limit it.

Reply:
Fixed